vars/transportRequestRelease.groovy

```diff
@@ -0,0 +1,65 @@
+import com.sap.piper.GitUtils
+import groovy.transform.Field
+
+import com.sap.piper.ConfigurationMerger
+import com.sap.piper.cm.ChangeManagement
+import com.sap.piper.cm.ChangeManagementException
+
+import hudson.AbortException
+
+
+@Field def STEP_NAME = 'transportRequestRelease'
+
+@Field Set parameterKeys = [
+    'changeId',
+    'transportRequestId',
+    'cmCredentialsId',
+    'cmEndpoint'
+  ]
+
+@Field Set stepConfigurationKeys = [
+    'cmCredentialsId',
+    'cmEndpoint'
+  ]
+
+def call(parameters = [:]) {
+
+    handlePipelineStepErrors (stepName: STEP_NAME, stepParameters: parameters) {
+
+        def script = parameters?.script ?: [commonPipelineEnvironment: commonPipelineEnvironment]
+
+        ChangeManagement cm = new ChangeManagement(script)
+
+        Map configuration = ConfigurationMerger.merge(script, STEP_NAME,
+                                                      parameters, parameterKeys,
+                                                      stepConfigurationKeys)
+
+        def changeId = configuration.changeId
```

Comment:
See above: we should be able to retrieve the ChangeId also from the git commits.

Reply:
I think that the `changeId` should be retrieve on a previous step, and it should be passed as parameter to this step.

Reply:
I suggest to keep the strategy the same for all cm releated steps. Nothing against the option to pass the id via signature, but the other ways (optional commonPipelineEnvironment (caching)) or retrieving from the changes should work here, too.

It is hard to imagine that releasing of a transport request is the only step in a pipeline without prior other cm related steps, but why should we intentionally disable a single-use?